keps/sig-auth/2907-secrets-store-csi-driver/README.md

```diff
@@ -0,0 +1,368 @@
+#### Authenticating to external secret APIs
+
+Authentication and authorization is largely up to the external API and its provider process, but the driver itself does include a few features that enable scoping access to secrets to individual pods.
+
+[Pod info](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/603-csi-pod-info) is propagated to the external provider.
+
+Additionally [KEP 1855](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1855-csi-driver-service-account-token) allows the driver to propagate a token to the provider. This enabled the providers to impersonate the pod for fetching secrets without the need granting large RBAC scopes.
+
+## Design Details
+
+### Test Plan
+
+- Automated pre and post submit end-to-end integration tests
+- Periodic end-to-end integration tests
+- Supported providers each have their own integration test suites along with a reference provider
+
+### Graduation Criteria
+
+#### GA
+
+- Month+ soak of minor release
+- Completion of milestone requirements
+- Agreement of stability documented on community call from 3+ provider maintainers
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+This will be deployed to clusters as a standalone separate component.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+
+###### How can someone using this feature know that it is working for their instance?
```

Comment:
how about liveness-probe?

Reply:
The liveness probe indicates the driver is up and running on the cluster but it doesn't provide more insight on the success/error mounting volumes for the pods. The metrics provide that info. We can also add successful creation of pod with volume here. WDYT?

Reply:
> We can also add successful creation of pod with volume here. WDYT?


sgtm

Maybe the liveness-probe can be added to this section: https://github.com/kubernetes/enhancements/pull/2939/files#diff-e05949ad35ba22ff89f499b278cbab33689e418a93228484824743ce56d5ca7fR229

Reply:
Added liveness-probe in `What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?`